lib/API/Oauth.dart

```diff
@@ -233,13 +207,13 @@ abstract class Auth {
 
     // Check if the token is not expired
     // if (_token != null) {
-    if ((_token != null) || (_token == "null")) {
+    if (_token != null && _token != "null") {
       globals.displayInfo(
           'token has been stored before! ${tokenStored.accessToken}  exp. ${tokenStored.expiresAt}');
     }
 
     // Use the refresh token to get a new access token
-    if (isExpired && ((_token != "null") || (_token != null))) {
+    if (isExpired && _token != null && _token != "null") {
```

Comment:
Here if either token is null or is "null" then the right hand side of the logical and becomes true. I feel like that the logical OR should be an AND because we want to test for the condition of the token is not null (which can have two variants). Bunch of extra parantheses removed.